omc3/tbt/handler.py

```diff
@@ -32,6 +32,28 @@ def __init__(self, matrices, date, bunch_ids, nturns):
         self.bunch_ids = bunch_ids
 
 
+def generate_average_tbtdata(tbtdata):
+    '''Takes a TbtData object and returns TbtData object containing the average
+    over all bunches/particles at all used BPMs '''
+    data = tbtdata.matrices
+    bpm_names = data[0]['X'].index
+
+    matrices = [{plane: pd.DataFrame(index=bpm_names,
+                                     data=get_averaged_data(bpm_names, data, plane, tbtdata.nturns),
+                                     dtype=float) for plane in PLANES}]
+    return TbtData(matrices, tbtdata.date, [1], tbtdata.nturns)
+
+
+def get_averaged_data(bpm_names, data, plane, turns):
+
+    bpm_data = np.zeros((len(bpm_names), len(data), turns))
```

Comment:
This will not work if some of the particles are lost

Reply:
fixed with last commit